src/redis.js

```diff
@@ -0,0 +1,61 @@
+var redis = require('redis');
+var debug = require('debug')('feathers-sync:redis');
+
+var pub = redis.createClient();
+var sub = redis.createClient();
+
+module.exports = function(config) {
+  debug('setting up database %s', config.db);
+
+  return function() {
+
+    var oldSetup = this.setup;
+
+    this.setup = function() {
+
+      var result = oldSetup.apply(this, arguments);
+      var services = this.services;
+
+      Object.keys(services).forEach(function(path) {
+
+        var service = services[path];
+        service._serviceEvents.forEach(function(event) {
+
+          var ev = path + ' ' + event;
+          debug('subscribing to handler %s', ev);
+
+          sub.subscribe(ev);
+          sub.on('message', function(data) {
+            if(data !== ev ){ return; }
+            debug('got event, calling old emit %s', data);
+            service._emit.call(service, event, data);
+          });
```

Comment:
Socketio should be available in `app.io` at this point. However, I think using Redis directly makes more sense since this will also work with other websocket libraries that Primus supports.

I see what you are saying with the subscription. I'll have to set it up locally to see if it is possible but could we flip it around maybe?